api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -253,7 +254,21 @@ message CertificateProviderPluginInstance {
   string certificate_name = 2;
 }
 
-// [#next-free-field: 15]
+// String matcher for subject alternative names, to match both type and value of the SAN.
+message SubjectAltNameMatcher {
+  enum SanType {
+    EMAIL = 0;
+    DNS = 1;
+    URI = 2;
+    IP_ADDRESS = 3;
+  }
+
+  SanType san_type = 1 [(validate.rules).enum = {defined_only: true}];
+
+  type.matcher.v3.StringMatcher matcher = 2;
```

Comment:
I think this can be annotated with `[(validate.rules).message = {required: true}]`.
Can you also add some comments to the fields.

Reply:
Modified existing rule to exclude default value. Added comments.

Reply:
IIUC this allows a config with a defined `san_type`, but without any `matcher`. Is that the intent?
If not, consider adding the PGV rule: `[(validate.rules).message = {required: true}]` to this field